compiler/src/parsing/lexer.re

```diff
@@ -386,77 +350,70 @@ let rec token = lexbuf => {
   | "@" => positioned(AT)
   | "b\"" =>
     let (start_p, _) = Sedlexing.lexing_positions(lexbuf);
-    read_str(start_p, Buffer.create(16), false, lexbuf);
+    let buf = Buffer.create(16);
+    Buffer.add_string(buf, Sedlexing.Utf8.lexeme(lexbuf));
+    read_bytes(start_p, buf, lexbuf);
   | '"' =>
     let (start_p, _) = Sedlexing.lexing_positions(lexbuf);
-    read_str(start_p, Buffer.create(16), true, lexbuf);
+    let buf = Buffer.create(16);
+    Buffer.add_string(buf, Sedlexing.Utf8.lexeme(lexbuf));
+    read_str(start_p, buf, lexbuf);
   | "'" =>
     let (start_p, _) = Sedlexing.lexing_positions(lexbuf);
-    read_char(start_p, Buffer.create(4), lexbuf);
+    let buf = Buffer.create(4);
+    Buffer.add_string(buf, Sedlexing.Utf8.lexeme(lexbuf));
+    read_char(start_p, buf, lexbuf);
   | "_" => positioned(UNDERSCORE)
   | lident => positioned(LIDENT(Sedlexing.Utf8.lexeme(lexbuf)))
   | uident => positioned(UIDENT(Sedlexing.Utf8.lexeme(lexbuf)))
   | eof => positioned(EOF)
   | _ => raise(Error(lexbuf_loc(lexbuf), UnrecognizedToken))
   };
 }
-and read_str = (start_p, buf, unicode, lexbuf) => {
+and read_bytes = (start_p, buf, lexbuf) => {
   switch%sedlex (lexbuf) {
-  | ('\\', newline_char) => read_str(start_p, buf, unicode, lexbuf)
-  | "\\b" =>
-    Buffer.add_char(buf, '\b');
-    read_str(start_p, buf, unicode, lexbuf);
-  | "\\f" =>
-    Buffer.add_char(buf, '\012');
-    read_str(start_p, buf, unicode, lexbuf);
-  | "\\n" =>
-    Buffer.add_char(buf, '\n');
-    read_str(start_p, buf, unicode, lexbuf);
-  | "\\r" =>
-    Buffer.add_char(buf, '\r');
-    read_str(start_p, buf, unicode, lexbuf);
-  | "\\t" =>
-    Buffer.add_char(buf, '\t');
-    read_str(start_p, buf, unicode, lexbuf);
-  | "\\v" =>
-    Buffer.add_char(buf, '\011');
-    read_str(start_p, buf, unicode, lexbuf);
-  | "\\\"" =>
-    Buffer.add_char(buf, '"');
-    read_str(start_p, buf, unicode, lexbuf);
+  // Eat all `\\` before `\"` to make sure it's not an escaped slash
```

Comment:
I want to drop the word `before` here since I first read this in the context of the lexing engine rather than in the string and got confused.
```suggestion
  // Eat all `\\` to ensure we handle `\"` correctly
```